src/main/java/io/tiledb/java/api/Condition.java

```diff
@@ -0,0 +1,88 @@
+package io.tiledb.java.api;
+
+import io.tiledb.libtiledb.*;
+import java.lang.reflect.Array;
+import java.math.BigInteger;
+
+public class Condition implements AutoCloseable {
+  private Context ctx;
+  private SWIGTYPE_p_tiledb_query_condition_t conditionp;
+  private SWIGTYPE_p_p_tiledb_query_condition_t conditionpp;
+  private Datatype type;
+
+  protected Condition(Context ctx, SWIGTYPE_p_p_tiledb_query_condition_t conditionpp) {
+    this.ctx = ctx;
+    this.conditionp = tiledb.tiledb_query_condition_tpp_value(conditionpp);
+    this.conditionpp = conditionpp;
+  }
+
+  protected Condition(
+      Context ctx,
+      String attributeName,
+      Object value,
+      Class conditionType,
+      tiledb_query_condition_op_t OP)
+      throws TileDBError {
+    this.type = Types.getNativeType(conditionType);
+    conditionpp = tiledb.new_tiledb_query_condition_tpp();
+    ctx.handleError(tiledb.tiledb_query_condition_alloc(ctx.getCtxp(), conditionpp));
```

Comment:
Let's put this to a `try` clause to handle the potential `TileDBError` exception

Reply:
Done!